js/config.js

```diff
@@ -214,128 +217,98 @@ function saveUserConfig(fieldName, newValue, oldConfig) {
-        // Picking some values from global config to overwrite user config
-        const configDataToUpdate = pick(newGlobalConfig, ['url', 'minimizeOnClose', 'launchOnStartup', 'alwaysOnTop']);
-        const updatedUserConfigData = Object.assign(oldUserConfig, configDataToUpdate);
-        const jsonNewConfig = JSON.stringify(updatedUserConfigData, null, ' ');
-        // get user config path
-        let userConfigFile;
+function updateUserConfig(oldUserConfig) {
+
+    return new Promise((resolve) => {
 
+        // create a new object from the old user config
+        // by ommitting the user related settings from
+        // the old user config
+        let newUserConfig = omit(oldUserConfig, ignoreSettings);
+        let newUserConfigString = JSON.stringify(newUserConfig, null, 4);
+
+        // get the user config path
+        let userConfigFile;
         if (isMac) {
             userConfigFile = path.join(dirs.userConfig(), configFileName);
         } else {
             userConfigFile = path.join(app.getPath('userData'), configFileName);
         }
 
-        fs.writeFile(userConfigFile, jsonNewConfig, 'utf8', (err) => {
-            if (err) {
-                reject(err);
-                return;
-            }
-            resolve();
-        });
+        // write the new user config changes to the user config file
+        fs.writeFileSync(userConfigFile, newUserConfigString, 'utf-8');
+
+        return resolve();
+
     });
+
 }
 
 /**
- * Method to overwrite user config on windows installer
- * @param {String} perUserInstall - Is a flag to determine whether we are installing for per user
+ * Manipulates user config on windows
+ * @param {String} perUserInstall - Is a flag to determine if we are installing for an individual user
  * @returns {Promise}
  */
 function updateUserConfigWin(perUserInstall) {
+
     return new Promise((resolve, reject) => {
+
+        // we get the user config path using electron
         const userConfigFile = path.join(app.getPath('userData'), configFileName);
-        // flag to determine whether per user installation
-        if (!perUserInstall) {
-            reject();
-            return;
-        }
 
-        // if user config file does't exists just copy global config file
-        if (!fs.existsSync(userConfigFile)) {
-            resolve(copyConfigWin());
+        // if it's not a per user installation or if the
+        // user config file doesn't exist, we simple move on
+        if (!perUserInstall || !fs.existsSync(userConfigFile)) {
+            log.send(logLevels.WARN, 'config: Could not find the user config file!');
+            reject();
             return;
         }
 
-        Promise.all([readGlobalConfig(), readUserConfig(userConfigFile)])
+        // In case the file exists, we remove it so that all the
+        // values are fetched from the global config
+        // https://perzoinc.atlassian.net/browse/ELECTRON-126
+        Promise.all([readUserConfig(userConfigFile)])
```

Comment:
do you need Promise.all here since readUserConfig should return a promise?

Reply:
Not really, I can remove it. Since we had multiple functions being called before and for future purposes, just left it as is.

I'll remove it for now.